gnovm/pkg/gnopkgfetch/gnopkgfetch.go

```diff
@@ -0,0 +1,130 @@
+package gnopkgfetch
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/gnolang/gno/gnovm/pkg/gnoimports"
+	"github.com/gnolang/gno/gnovm/pkg/gnolang"
+	"github.com/gnolang/gno/gnovm/pkg/gnomod"
+	tm2client "github.com/gnolang/gno/tm2/pkg/bft/rpc/client"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/gnolang/gno/tm2/pkg/errors"
+	"golang.org/x/mod/module"
+)
+
+// FetchPackageImportsRecursively recursively fetches the imports of a local package while following a given gno.mod replace directives
+func FetchPackageImportsRecursively(io commands.IO, pkgDir string, gnoMod *gnomod.File) error {
```

Comment:
Please consider not having `commands.IO` as an argument;
If you need to print, handle it at the caller level in the command itself

Reply:
Since this package is now a `gno/cmd` internal, is it acceptable to keep it this way?

We don't know what to print from the caller:
- we don't know the full dependency tree as it's discovered when packages are downloaded
- lines are printed as packages start to download
- if a package is already cached, it's download line is not printed
this mimicks go commands behavior

We could pass an `io.Writer` from `commands.IO.Err` instead of `commands.IO` directly but I think it's worse since `commands.IO` then looses control over how things are printed and there is no benefit IMO.